extensions/last-commit-info/src/main/java/de/sovity/edc/extension/LastCommitInfoService.java

```diff
@@ -16,42 +16,56 @@
 
 import org.eclipse.edc.spi.system.ServiceExtensionContext;
 
+import java.io.IOException;
 import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Objects;
 import java.util.Scanner;
+import java.util.jar.Attributes;
+import java.util.jar.Manifest;
 
 public class LastCommitInfoService {
 
     private final ServiceExtensionContext context;
 
+
     public LastCommitInfoService(ServiceExtensionContext context) {
         this.context = context;
     }
 
-    public String getLastCommitInfo() {
-        var result = "";
-
-        if (!getEnvLastCommitInfo().equals("")) {
-            result += "Env Last Commit Info: \n";
-            result += getEnvLastCommitInfo() + "\n";
-        }
-
-        if (!getEnvLastCommitInfo().equals(getJarLastCommitInfo())) {
-            result += "Jar Last Commit Info: \n";
-            result += getJarLastCommitInfo();
-        }
+    public LastCommitInfo getLastCommitInfo() throws IOException {
+        var result = new LastCommitInfo();
+        result.setEnvLastCommitInfo(getEnvLastCommitInfo());
+        result.setEnvLastBuildDate(getEnvLastCommitDate());
 
+        result.setJarLastCommitInfo(getJarInfo().get(0));
+        result.setJarLastBuildDate(getJarInfo().get(1));
         return result;
     }
 
-    public String getJarLastCommitInfo() {
+    public List<String> getJarInfo() throws IOException {
+
+        var jarInfo = new ArrayList<String>();
         var classLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
this should be replaced with LastCommitInfoService.class.getClassLoader() or getContextClassLoader() whichever one exists.

we want to read the files in the jar where this class lies

using hte thread context class loader does not read the files in this jar